config.go

```diff
@@ -0,0 +1,526 @@
+// Copyright (C) 2013-2017 The btcsuite developers
+// Copyright (C) 2015-2016 The Decred developers
+// Copyright (C) 2015-2022 Lightning Labs and The Lightning Network Developers
+// Copyright (C) 2022 Bottlepay and The Lightning Network Developers
+
+package lndsigner
+
+import (
+	"encoding/hex"
+	"fmt"
+	"net"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg"
+	flags "github.com/jessevdk/go-flags"
+)
+
+const (
+	defaultConfigFilename  = "signer.conf"
+	defaultTLSCertFilename = "tls.cert"
+	defaultTLSKeyFilename  = "tls.key"
+	defaultLogLevel        = "info"
+	defaultLogDirname      = "logs"
+	defaultLogFilename     = "signer.log"
+	defaultRPCPort         = 10009
+	defaultRPCHost         = "localhost"
+
+	defaultMaxLogFiles    = 3
+	defaultMaxLogFileSize = 10
```

Comment:
Something to check is if we need logs files on disk at all, or that we'll assume the infra takes care of it.

One other changing with regards to logging is that we could move to `zap.Logger` for json logging.